src/components/Hidayah/index.js

```diff
@@ -17,12 +19,50 @@ const PopUpModal = ({ popUpClassName, displayPopUpModal, setDisplayPopUpModal, p
   </div>);
 };
 
+const Levels = ({ level, setLevel }) => (<div>
+  { LEVELS && LEVELS.map((lvl) => (
+    <>
+      <input type="radio" id={lvl} name={lvl} value={lvl} checked={lvl === level} onChange={() => setLevel(lvl)} />
+      <label key={lvl} for={lvl}>{lvl}</label>
+    </>
+  ))}
+</div>);
+
+const SettingsModal = ({ settings, setSettings, hideModal, settingsModalClassName }) => (<div style={{ position: 'absolute', top: 0, left: 0, width: '100%', height: '100%', padding: '1%', backgroundColor: '#ddd' }} className={settingsModalClassName || ''}>
+  <button style={{ position: 'absolute', top: '1%', right: '1%', zIndex: 1 }} onClick={hideModal}>X</button>
+  <h3>Settings</h3>
+  <h4>Levels:</h4>
+  <Levels level={settings.level} setLevel={(newLevel) => setSettings((old) => {
```

Comment:
this too should accept a client function that can translate these levels

Reply:
Won't that be a lot of settings to be config from the client side ?